src/static-build/loader.ts

```diff
@@ -0,0 +1,155 @@
+import getFeatures from './getFeatures';
+import { LoaderContext } from 'webpack/lib/webpack';
+const { getOptions } = require('loader-utils');
+import * as recast from 'recast';
+const types = recast.types;
+const namedTypes = types.namedTypes;
+const builders = types.builders;
+const compose = require('recast/lib/util').composeSourceMaps;
+
+/**
+ * A map of features that should be statically replaced in the code
+ */
+export interface StaticHasFeatures {
+	[feature: string]: boolean;
+}
+
+const HAS_MID = /\/has$/;
+const HAS_PRAGMA = /^\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*$/;
+const HAS_MID_SNIFF = /require\(["'][^,"']+\/has["']\)/;
+const HAS_PRAGMA_SNIFF = /^\s*['"]\s*(!?)\s*has\s*\(["']([^'"]+)['"]\)\s*['"]\s*;\s*$/m;
+
+export default function (this: LoaderContext, content: string, sourceMap?: { file: '' }) {
```

Comment:
We should name this function (maybe `loader`) and provide some JSDoc.

The type annotation of `{ file: '' }` doesn't make sense.  I would prefer we properly type that (which is I believe a source-map RawSourceMap?) or at least not use a string literal of `''`.

We also should annotate the return type on an exported/public function as this helps ensure that all of our returns match the desired API versus them just being implicit.